File: app/src/main/java/com/wolcano/musicplayer/music/mvp/interactor/interfaces/AlbumInteractor.java
package com.wolcano.musicplayer.music.mvp.interactor.interfaces;

import android.app.Activity;

import com.wolcano.musicplayer.music.mvp.models.Album;

import java.util.List;

import io.reactivex.disposables.Disposable;

public interface AlbumInteractor {

    interface OnGetAlbumListener{
        void sendAlbum(List<Album> albumList);
        void controlIfEmpty();
    }

    void getAlbum(Activity activity, Disposable disposable, String sort, OnGetAlbumListener onGetAlbumListener);

}

File: app/src/main/java/com/wolcano/musicplayer/music/mvp/interactor/interfaces/AlbumInteractor.kt
package com.wolcano.musicplayer.music.mvp.interactor.interfaces

import android.app.Activity
import com.wolcano.musicplayer.music.mvp.models.Album

interface AlbumInteractor {

    interface OnGetAlbumListener {
        fun sendAlbum(albumList: ArrayList<Album>?)
        fun controlIfEmpty()
    }

    fun getAlbum(activity: Activity?, sort: String?, onGetAlbumListener: OnGetAlbumListener?)

}